src/windshaft-integration/model-updater.js

```diff
@@ -266,4 +268,27 @@ ModelUpdater.prototype._getLayerLegends = function (layerModel) {
     layerModel.legends.choropleth
   ];
 };
+
+/**
+ * Return the analysis list without duplicates
+ * 
+ * (a dataview and a layer could share some analysis)
+ */
+ModelUpdater.prototype._getUniqueAnalyses = function () {
```

Comment:
Perhaps we should differentiate between the terms "Analysis" and "Analysis node" in the code (we're already doing it in the body of this method). Since this method is returning "nodes", perhaps it should reflect that in its name, so I'd rename it to something like `_getAnalysisNodes`.